drivers/SmartThings/matter-appliance/src/matter-dishwasher/init.lua

```diff
@@ -95,6 +102,9 @@ end
 local function dishwasher_supported_modes_attr_handler(driver, device, ib, response)
   dishwasherModeSupportedModes = {}
   for _, mode in ipairs(ib.data.elements) do
+    if version.api < 10 then
```

Comment:
This is only done for older API versions because on hubs where they already have the 53 lua libs with Matter 1.2 support, then when this struct is deserialized, it will be augmented to use the appropriate type with the correct field defs when it is deserialized. However, without the 53 lua libs, it is just deserialized as a base struct data type and is not augemented to the SupportedModes type. It has all the same data in the same structure, but the struct is just missing the field defs, so augmenting here allows us to access the values of the struct using the field names rather than index directly to the value with a hardcoded number.